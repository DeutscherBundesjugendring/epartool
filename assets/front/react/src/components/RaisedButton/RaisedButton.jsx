import React from 'react';
import PropTypes from 'prop-types';

const RaisedButton = ({
  disabled,
  id,
  label,
  onClick,
}) => {
  const linkClasses = 'btn btn-default btn-sm';

  return (
    <button
      id={id}
      type="button"
      onClick={() => onClick()}
      disabled={disabled}
      className={linkClasses}
    >
      {label}
    </button>
  );
};

RaisedButton.defaultProps = {
  disabled: false,
  id: null,
};

RaisedButton.propTypes = {
  disabled: PropTypes.bool,
  id: PropTypes.string,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default RaisedButton;
