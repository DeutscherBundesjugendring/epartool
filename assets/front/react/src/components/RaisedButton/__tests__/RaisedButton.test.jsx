import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import RaisedButton from '../RaisedButton';


injectTapEventPlugin();

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <RaisedButton label="text" onTouchTap={() => {}} />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly disabled button', () => {
    const tree = shallow(
      <RaisedButton label="text" onTouchTap={() => {}} disabled />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls onTouchTap', () => {
    const spy = sinon.spy();
    const component = shallow(
      <RaisedButton label="text" onTouchTap={spy} />
    );

    component.simulate('touchTap', { stopPropagation: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });
});
