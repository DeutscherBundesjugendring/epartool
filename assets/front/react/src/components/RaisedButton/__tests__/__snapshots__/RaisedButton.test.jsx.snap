// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<button
  className="btn btn-default"
  disabled={false}
  onTouchTap={[Function]}
>
  text
</button>
`;

exports[`rendering renders correctly disabled button 1`] = `
<button
  className="btn btn-default"
  disabled={true}
  onTouchTap={[Function]}
>
  text
</button>
`;
